relay-event-normalization/src/normalize/span/description/sql/parser.rs

```diff
@@ -0,0 +1,274 @@
+    fn transform_query(query: &mut Query) {
+        if let SetExpr::Select(select) = &mut *query.body {
+            // Track collapsible selected items (e.g. `SELECT col1, col2`).
+            let mut collapse = vec![];
+
+            // Iterate over selected item.
+            for mut item in std::mem::take(&mut select.projection) {
+                // Normalize aliases.
+                if let SelectItem::ExprWithAlias { ref mut alias, .. } = &mut item {
+                    alias.quote_style = None;
+                }
+                if Self::is_collapsible(&item) {
+                    collapse.push(item);
+                } else {
+                    Self::collapse_items(&mut collapse, &mut select.projection);
+                    collapse.clear();
+                    select.projection.push(item);
+                }
+            }
+            Self::collapse_items(&mut collapse, &mut select.projection);
+
+            // Iterate "FROM".
+            for from in select.from.iter_mut() {
+                match &mut from.relation {
+                    // Recurse into subqueries.
+                    TableFactor::Derived { subquery, .. } => {
+                        Self::transform_query(subquery);
+                    }
+                    // Strip quotes from identifiers in table names.
+                    TableFactor::Table { name, .. } => {
+                        for ident in &mut name.0 {
+                            ident.quote_style = None;
+                        }
+                    }
+                    _ => (),
+                }
+            }
+        }
+    }
+
+    fn simplify_compound_identifier(parts: &mut Vec<Ident>) {
+        if let Some(mut last) = parts.pop() {
+            last.quote_style = None;
+            *parts = vec![last];
+        }
+    }
+}
+
+impl VisitorMut for NormalizeVisitor {
+    type Break = ();
+
+    fn pre_visit_expr(&mut self, expr: &mut Expr) -> ControlFlow<Self::Break> {
+        match expr {
+            // Simple values like numbers and strings are replaced by a placeholder:
+            Expr::Value(x) => *x = Self::placeholder(),
+            // Casts are omitted for simplification.
+            Expr::Cast { expr: inner, .. } => {
+                *expr = *inner.clone(); // clone is unfortunate here.
+            }
+            // `IN (val1, val2, val3)` is replaced by `IN (%s)`.
+            Expr::InList { list, .. } => *list = vec![Expr::Value(Self::placeholder())],
+            // `"table"."col"` is replaced by `col`.
+            Expr::CompoundIdentifier(parts) => {
+                Self::simplify_compound_identifier(parts);
+            }
+            // `"col"` is replaced by `col`.
+            Expr::Identifier(ident) => {
+                ident.quote_style = None;
+            }
+            // Recurse into subqueries.
+            Expr::Subquery(query) => Self::transform_query(query),
+            // Remove negative sign, e.g. `-100` becomes `%s`.
```

Comment:
IIUC this doesn't remove the negative sign, but just resets the expression (i.e. `-100` becomes `-%s`)? See https://github.com/sqlparser-rs/sqlparser-rs/blob/4c3a4ad5a87ef8b4dd24a6e8540243e14fbfa148/src/ast/mod.rs#L729